src/ReferenceList.php

```diff
@@ -59,31 +62,23 @@ public function __construct( $references = array() ) {
 	 * @throws InvalidArgumentException
 	 */
 	public function addReference( Reference $reference, $index = null ) {
-		if ( !is_int( $index ) && $index !== null ) {
-			throw new InvalidArgumentException( '$index must be an integer or null' );
+		if ( $index !== null && ( !is_int( $index ) || $index < 0 ) ) {
```

Comment:
An other breaking change that must be documented. There should also be a test for this.